cmd/hc/main.go

```diff
@@ -41,7 +41,10 @@ var (
 		Overlay   overlay.Config
 	}
 	setupCfg struct {
-		BasePath string `default:"$CONFDIR" help:"base path for setup"`
+		BasePath    string `default:"$CONFDIR" help:"base path for setup"`
+		Concurrency uint   `default:"4" help:"number of concurrent workers for certificate authority generation"`
+		CA          provider.CAConfig
+		Identity    provider.IdentityConfig
```

Comment:
why are these last two here? CAConfig defines paths, a difficulty, and a version, but all of those should be calculable from BasePath, Concurrency, and setup should just choose the latest version. the idea behind setupCfg is everything you need to generate a good config.

Reply:
What if you want to specify separate paths for the key vs cert for example?